src/stock-adjustment-mobile/add-products-page/add-product-page.jsx

```diff
@@ -0,0 +1,265 @@
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import React, { useMemo, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import {useDispatch, useSelector} from 'react-redux';
+import { Form } from 'react-final-form';
+import arrayMutators from 'final-form-arrays';
+import { FieldArray } from 'react-final-form-arrays';
+import createDecorator from 'final-form-calculate';
+
+import InputField from '../../react-components/form-fields/input-field';
+import SelectField from '../../react-components/form-fields/select-field';
+import ReadOnlyField from '../../react-components/form-fields/read-only-field';
+import { formatLot, formatDate, formatDateISO } from '../format-utils';
+import AddButton from '../../react-components/buttons/add-button';
+import { appendToAdjustment } from '../reducers/adjustment';
+
+
+const AddProductsPage = ({}) => {
+    const history = useHistory();
+    const dispatch = useDispatch();
+    
+    const userHomeFacility = useSelector(state => state.facilities.userHomeFacility);
+    const productOptions = useSelector(state => state.productOptions.productOptions);
+    const reasons = useSelector(state => state.reasons.reasons);
+    const adjustment = useSelector(state => state.adjustment.adjustment);
+    const program = useSelector(state => state.program.program);
+
+    const decorator = useMemo(() => createDecorator(
+    {
+        field: /product/,
+        updates: {
+            stockOnHand: (productVal, itemsVal) => {
+                const orderable = itemsVal.items[0].hasOwnProperty('product') ? itemsVal.items[0].product : []; 
+                if(itemsVal.items[0].hasOwnProperty('lot')) {
+                    delete itemsVal.items[0].lot;
+                } 
+                const lotCode = null; 
+                const stockOnHand = getStockOnHand(orderable, lotCode);
+                return stockOnHand;
+            }
+        }
+    },
+    {
+        field: /lot/,
+        updates: {
+            stockOnHand: (productVal, itemsVal) => {
+                const orderable = itemsVal.items[0].hasOwnProperty('product') ? itemsVal.items[0].product : []; 
+                const lotCode = itemsVal.items[0].hasOwnProperty('lot') ? itemsVal.items[0].lot.lotCode : null; 
+                const stockOnHand = getStockOnHand(orderable, lotCode);
+                return stockOnHand;
+            }
+        }
+    }
+    ), []);
+
+    const isQuantityNotFilled = (quantity) => {
+        return _.isUndefined(quantity) || _.isNull(quantity) || _.isNaN(quantity) || quantity === "";
```

Comment:
Yes. I agree, But I've just checked this on current local build and there is error with 'Uncaught TypeError: _.isNil is not a function'. It could be related to 'Lodash' version. I would rather not touch current version.